Mod interact viz

ryanpeek opened the pull request:
Ok, let know what you think. I kept this one a bit simpler and shorter. I'm sure there are other options but wanted this to be a short fun lesson to just show a few common options/tools. It's a little bare, so if you see spots to add an `aside` or two, feel free. I wanted to make some progress and not let perfect be the enemy of done. 😉 

I'm only pushing up the `Rmd` because this has a number of supporting `site_libs` that are part of leaflet/D3...so probably best to just save the final push for the `main` branch. If you have trouble knitting it let me know and I can push everything up.

richpauloo commented:
This is awesome and I love that the geocoder and mapview came to life! "Hiding from fox" is my favorite. lolz

I changed the "address" colname to make it clearer that this was a restaurant address, not a work or personal address--something like "restaurant_address". Can you also add a field to the form above the address called, "restaurant name"? I think that, combined with the address will make the viz really speak!

Perhaps it's just my config, but when I knit, the leaflet doesn't show the markers. I have to toggle the groups "home" and "food and drink" for them to show on the map. Does that happen with you? 

Also I changed many "1-page" to "l-page" which I think was your intent in getting the figures to occupy more space.

Finally, in some places you wrap an object assignment for a figure in parenthesis like so `( p <- ggplot(data, aes(x,y)) + geom_point() )`. It might be worthwhile to add a quick aside the first time that happens to explain what that does.

ryanpeek commented:
Good suggestions...thanks. I updated the form and col names as suggested. Makes sense. Also added an aside. I had a similar issue with the leaflet, where I had to click the layer on and off to get it to appear...I think for now I'm ok with it at least being functional. Not sure what is causing this, but we can trouble shoot later...(or just use this as a good "teachable moment" ;)

Let me know if you are ok with merging this now.
